dev/system-test/firestore.ts

```diff
@@ -1670,12 +1768,53 @@ describe('Query class', () => {
     expectDocs(res, {doc: 2}, {doc: 3}, {doc: 4});
   });
 
-  it('has offset() method', async () => {
+  it('has offset() method on get()', async () => {
     await addDocs({foo: 'a'}, {foo: 'b'});
     const res = await randomCol.orderBy('foo').offset(1).get();
     expectDocs(res, {foo: 'b'});
   });
 
+  it('has offset() method on stream()', done => {
+    let received = 0;
+    Promise.all([addDocs({foo: 'a'}, {foo: 'b'})]).then(() => {
+      return randomCol
+        .orderBy('foo')
+        .offset(1)
+        .stream()
+        .on('data', doc => {
+          expect(doc.data()).to.deep.equal({foo: 'b'});
+          ++received;
+        })
+        .on('end', () => {
+          expect(received).to.equal(1);
+          done();
+        });
+    });
+  });
+
+  it('can run offset(num), where num is larger than the collection size on get()', async () => {
+    await addDocs({foo: 'a'}, {foo: 'b'});
```

Comment:
I prefer this use of async/await style over the Promise+done style that you used in other tests. It's easier to read. But the Promise+done style was used in existing tests in the file, so I didn't say anything. However, I'd recommend just being consistent across the new tests you're writing, whatever style you choose.